map/lanelet2_extension/lib/query.cpp

```diff
@@ -37,11 +38,10 @@ namespace
 {
 double getAngleDifference(const double angle1, const double angle2)
 {
-  Eigen::Vector2d vec1, vec2;
-  vec1 << std::cos(angle1), std::sin(angle1);
-  vec2 << std::cos(angle2), std::sin(angle2);
-  const double diff_angle = std::acos(vec1.dot(vec2));
-  return std::fabs(diff_angle);
+  const double normalized_angle1 = tier4_autoware_utils::normalizeRadian(angle1);
+  const double normalized_angle2 = tier4_autoware_utils::normalizeRadian(angle2);
+  const double diff_angle = std::fabs(normalized_angle1 - normalized_angle2);
+  return diff_angle;
```

Comment:
@kyoichi-sugahara cc @taikitanaka3 Actually, I think `return std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2));` is enough.
Also for usability, we can consider removing `std::abs`. 

I've said `If you use autoware_utils, I think adding tests isn't mandatory because it's already tested` before, but since it's not used as-is (std::abs is added), it's better to add tests to clarify the specification of this function.

Reply:
@kenji-miyake 
as he says 
> but still there are many functions which don't have test in the file so I will create a follow-up issue.
so I also think it would be better to add issues "making unit tests for each functions"

> std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2)); is enough.
substruction of not normalized value would more likely to return unintended value like pi - (-pi/0.9)

Reply:
@kenji-miyake 
as you said std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2)); would be better if @kyoichi-sugahara-san wants to consider normalized angle difference of both angle I tested here.
https://github.com/autowarefoundation/autoware.universe/pull/264

Reply:
@kenji-miyake 
If both angle1 and angle2 are normalized, then `return std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2));` is enough I think. But not sure the if output of `tf2::getYaw` is normalized or not. Do you know that?

Reply:
@kyoichi-sugahara 
if you want to assume pi and -pi is same angle below would be better 
`std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2));`
if you want to assume pi and -pi is 2pi different angle below would be better 
```
const double normalized_angle1 = tier4_autoware_utils::normalizeRadian(angle1);
const double normalized_angle2 = tier4_autoware_utils::normalizeRadian(angle2);
const double diff_angle = std::fabs(normalized_angle1 - normalized_angle2);
```

Reply:
> If both angle1 and angle2 are normalized, then return std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2)); is enough

It depends on the use case.

> But not sure the if output of tf2::getYaw is normalized or not. Do you know that?

I'm not sure, but anyway, we should define the output spec of functions.

Reply:
@taikitanaka3 
first of all, assuming same angle for pi and -pi. 
but output of `normalizeRadian(pi)` is -pi so comparison of pi and -pi with code below outputs 0 so no problem.
```
const double normalized_angle1 = tier4_autoware_utils::normalizeRadian(angle1);
const double normalized_angle2 = tier4_autoware_utils::normalizeRadian(angle2);
const double diff_angle = std::fabs(normalized_angle1 - normalized_angle2);
```
but i think i was misunderstanding for this comment. 
> If both angle1 and angle2 are normalized, then return std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2)); is enough I think.

Even angle was not normalized, std::abs(tier4_autoware_utils::normalizeRadian(angle1 - angle2));` is enough as you tested.